Geometry/HGCalCommonData/src/HGCalCellUV.cc

```diff
@@ -0,0 +1,277 @@
+  double y = x * fsin[placement % 6] + yloc * fcos[placement % 6];
+  x = x * fcos[placement % 6] - yloc * fsin[placement % 6];
+
+  int32_t u(-100), v(-100);
+  int ncell = (type != 0) ? ncell_[1] : ncell_[0];
+  double r = (type != 0) ? cellY_[1] : cellY_[0];
+  double l1 = (y / r) + ncell - 1.0;
+  int l2 = std::floor((0.5 * y + 0.5 * x / sqrt3_) / r + ncell - 4.0 / 3.0);
+  int l3 = std::floor((x / sqrt3_) / r + ncell - 4.0 / 3.0);
+  double l4 = (y + sqrt3_ * x) / (2 * r) + 2 * ncell - 2;
+  double l5 = (y - sqrt3_ * x) / (2 * r) - ncell;
+  double u1 = (y / r) + ncell + 1.0;
+  int u2 = std::ceil((0.5 * y + 0.5 * x / sqrt3_) / r + ncell + 2.0 / 3.0);
+  int u3 = std::ceil((x / sqrt3_) / r + ncell);
+  double u4 = l4 + 2;
+  double u5 = l5 + 2;
+
+  for (int ui = l2 + 1; ui < u2; ui++) {
+    for (int vi = l3 + 1; vi < u3; vi++) {
+      int c1 = 2 * ui - vi;
+      int c4 = ui + vi;
+      int c5 = ui - 2 * vi;
+      if ((c1 < u1) && (c1 > l1) && (c4 < u4) && (c4 > l4) && (c5 < u5) && (c5 > l5)) {
+        u = ui;
+        v = vi;
+      }
+    }
+  }
+
+  //--- Taking care of extending cells
+  if (v == -1) {
+    if (y < (2 * u - v - ncell) * r) {
+      v += 1;
+    } else {
+      u += 1;
+      v += 1;
+    }
+  }
+  if (v - u == ncell) {
+    if ((y + sqrt3_ * x) < ((u + v - 2 * ncell + 1) * 2 * r)) {
+      v += -1;
+    } else {
+      u += 1;
+    }
+  }
+  if (u == 2 * ncell) {
+    if ((y - sqrt3_ * x) < ((u - 2 * v + ncell - 1) * 2 * r)) {
+      u += -1;
+    } else {
+      u += -1;
+      v += -1;
+    }
+  }
+  if (debug)
+    edm::LogVerbatim("HGCalGeom") << "cellUVFromXY2: Input " << xloc << ":" << yloc << ":" << extend << " Output " << u
+                                  << ":" << v;
+  return std::make_pair(u, v);
+}
+
+std::pair<int32_t, int32_t> HGCalCellUV::cellUVFromXY3(
+    double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug) {
+  //--- Using Cube coordinates to find the (u, v)
+  //--- Reverse transform to placement=0, if placement index ≠ 6
+  double xloc1 = (placement >= 6) ? xloc : -xloc;
+  int rot = placement % 6;
+  const std::vector<double> fcos = {1.0, 0.5, -0.5, -1.0, -0.5, 0.5};
+  const std::vector<double> fsin = {0.0, sqrt3By2_, sqrt3By2_, 0.0, -sqrt3By2_, -sqrt3By2_};
+  double xprime = xloc1 * fcos[rot] - yloc * fsin[rot];
+  double yprime = xloc1 * fsin[rot] + yloc * fcos[rot];
+  double x = xprime + cellX_[type];
+  double y = yprime;
+
+  x = x / cellX_[type];
+  y = y / cellY_[type];
+
+  double cu = 2 * x / 3;
+  double cv = -x / 3 + y / 2;
+  double cw = -x / 3 - y / 2;
+
+  int iu = std::round(cu);
+  int iv = std::round(cv);
+  int iw = std::round(cw);
+
+  if (iu + iv + iw != 0) {
+    double arr[] = {std::abs(cu - iu), std::abs(cv - iv), std::abs(cw - iw)};
+    int i = std::distance(arr, std::max_element(arr, arr + 3));
+
+    switch (i) {
+      case 0:
+        iu = (std::round(cu) == std::floor(cu)) ? std::ceil(cu) : std::floor(cu);
```

Comment:
`iu` is not used any more in the following, no need to update it

Reply:
Removed the switch case for i=0 (where iu is updated, and using if-else instead of switch.